tools/brokenLinks.ts

```diff
@@ -0,0 +1,80 @@
+// DISCLAIMER: Only test this tool on mac
+// There may be some false negative results
```

Comment:
Why are we building a personal tool to check for broken links as opposed to using some kind of text or markdown linting?

And why does it only work on mac?

Reply:
it is NOT a personal tool, it is for all the people who have linux or macOS. 
it work for mac because I do not test on linux and I cannot claim that it works on linux :-)
it does not work for windows because in markdown `/` is used for path. we have to replace them with '\' inorder for windows OS to work.

Please let me know if you think that this tool is not use. I will not have it in git if your answer is yes.

thanks

Reply:
Sorry, by "personal",  I head meant a tool made specifically used for SPK codebase. I am fine with keeping it. But again, should we be looking into a text/markdown lint tool for our repo?

Reply:
i cannot find any lint tools for markdown hence I create this one (just to check for broken links)

Reply:
@dennisseah how does this tool alert us about future broken links?

Reply:
I guess my question is shouldn't we be running this as part of the CI, so that we see a build failure when there's a broken link?

Reply:
i do not know how can we do this. please provide suggestion. I have just written a simple tool to check for broken links and fix them. Please check with Andre if you need something more.